src/app/core/duplicate-records/duplicate-records.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+import { EntityMapperService } from '../entity/entity-mapper/entity-mapper.service'; 
+import { EntityRegistry } from '../entity/database-entity.decorator';
+import { EntitySchemaService } from '../entity/schema/entity-schema.service';
+import { Entity } from '../entity/model/entity';
+import { MatSnackBar, MatSnackBarHorizontalPosition, MatSnackBarVerticalPosition } from '@angular/material/snack-bar';
+import { ConfigService } from '../config/config.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class DuplicateRecordService {
+  get: jasmine.Spy<jasmine.Func>;
+  constructor(
+    private entitymapperservice: EntityMapperService,
+    private entityTypes: EntityRegistry,
+    private entityService: EntitySchemaService,
+    private snackBar: MatSnackBar,
+    private config: ConfigService
+  ) {}
+
+  showFlashMessage(message: string) {
+    const horizontalPosition: MatSnackBarHorizontalPosition = 'center';
+    const verticalPosition: MatSnackBarVerticalPosition = 'top';
+    this.snackBar.open(message, 'Close', {
+      duration: 3000, 
+      horizontalPosition: horizontalPosition,
+      verticalPosition: verticalPosition,
+    });
+  }
+
+  async duplicateRecord(sourceData: any, schemaName: string) {
+    if (!sourceData || sourceData.length === 0) {
```

Comment:
Should we hide the "duplicate" action from the menu when no items are selected?

Reply:
@sleidig I think we should show the "duplicate" action all the time. That way the user knows that there is an option to duplicate the records. The duplicate action itself looks a bit hidden under the 3 dots. Conditionally showing them would make it a bit difficult for the users to figure out themselves. Let us know your thougts.

Reply:
That's a valid point! In fact, I was also thinking how to make the action more visible at least whenever something is selected ... an animation or some element that is visible without opening the menu, possibly.

For now, I think keeping the option always visible, as you suggest, is good. I would then disable the button when nothing is selected and when disabled display a matTooltip explaining to first select some rows.

Reply:
Yes sounds good. @brajesh-lab let's update it accordingly.

Reply:
@rudresh @sleidig  I have updated it.

Reply:
Looks great :+1: